access-grant/src/main/java/com/inrupt/client/accessgrant/AccessCredentialVerification.java

```diff
@@ -28,9 +28,18 @@
  */
 public class AccessCredentialVerification {
 
-    private final List<String> checks;
-    private final List<String> warnings;
-    private final List<String> errors;
+    private List<String> checks;
+    private List<String> warnings;
+    private List<String> errors;
+
+    /**
+     * Create an empty verification response.
+     */
+    public AccessCredentialVerification() {
```

Comment:
This class is indeed only used for deserializing responses from the `/verify` endpoint. It was initially immutable, with a parametrized constructor and `final` members. However, this relied on `VerificationResponseData`, a class private to `AccessGrantClient`. Due to JSON-B introspection restrictions, `VerificationResponseData` should be public, which is undesirable. The current implementation of `AccessCredentialVerification` works around this issue as you describe, by exposing a default constructor (as required by JSON-B, since we aren't using annotations), and still making the members immutable once they have been initialized.

I can't tell if it is a hard requirement that `AccessCredentialVerification` instances are immutable. It makes sense semantically, but I don't know how crucial it is vs implementing less circonvoluted patterns.

Reply:
I see.

The class proposed here would not be truely immutable.
The collections its members return _are_ immutable, which is good.
However the collections _themselves_ can be overwritten with the new setters, which is not good.

How about the following pattern:
There is a class `VerificationResponseData` (as before). It's a pure DTO, a POJO with public, writable fields that is used for JSON deserialization only. That class itself must be public for JSONB, which is undesirable, as you say. We document it for internal user only.

The `AccessCredentialVerification` class, on the other hand, is a wrapper around the DTO (as before), that exposes only getters that make the fields immutable.
That way you return a truely immutable, clean API class.

The only disadvantage is exposing the DTO, but I think it's worth it. Maybe even suppress the warning for public fields and mutable collections.

Something like
```java
/**
 * for internal use only
 */
public class VerificationResponseData {
  /**
   * for internal use only
   */
  public List<String> checks;

  // etc
}

/**
 * properly documented
 */
public class AccessCredentialVerification {
  private final VerificationResponseData  original;

  /* not even public */ AccessCredentialVerification(final VerificationResponseData original) {
    this.original = original;
  }

  /**
   * properly documented
   */
  public List<String> getChecks() {
    return Collections.unmodifiableList(original.checks);
  }
}
```

Reply:
Based on the conversation in https://github.com/inrupt/solid-client-java/pull/542, my previous comment was mislead: it is not a hard requirement that the verification data class is immutable. I think exposing internals in the public API requires more care from us when refactoring the code, so let's just make the class mutable.
However, since the current gimmick of having immutable collections in mutable attributes doesn't really follow any well-established pattern, let's just make everything mutable for consistency, so that it isn't confusing.